modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -423,6 +417,28 @@ def _dtypes_for_exprs(self, exprs):
         """
         return [expr._dtype for expr in exprs.values()]
 
+    def set_dtypes_cache(self, dtypes):
+        """
+        Set dtypes cache.
+
+        Parameters
+        ----------
+        dtypes : pandas.Series, ModinDtypes or callable
+        """
+        if self._has_arrow_table() and self._partitions.size > 0:
```

Comment:
Is the partitions size check required here? The method _has_arrow_table() returns False in case of empty partitions.

Reply:
`_has_arrow_table` only checks that the `self._partitions` attribute is not None, meaning that we still can have empty partitions (`self._partitions == []`) and get `True` from this method:
https://github.com/modin-project/modin/blob/583036acf61973fa41b6f01ebce70c67dcd280df/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L407-L409

Note that calling `all` on an empty list returns True
```python
>>> all(x for x in [])
True
```

I've copied this `.size > 0` check from the dataframe's constructor, so apparently it's used for some purpose
https://github.com/modin-project/modin/blob/583036acf61973fa41b6f01ebce70c67dcd280df/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L237

Reply:
It seems like a bug and the size check should be added to the _has_arrow_table() method.